pkg/kudoctl/cmd/install/install.go

```diff
@@ -88,37 +83,59 @@ func installFrameworks(args []string, options *Options) error {
 	}
 
 	for _, name := range args {
-		err := installFramework(name, "", *r, indexFile, kc, options)
+		err := installFramework(name, "", *r, kc, options)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// installFramework is the umbrella for a single framework installation that gathers the business logic
-// for a cluster and returns an error in case there is a problem
-// TODO: needs testing
-func installFramework(name, previous string, repository repo.FrameworkRepository, indexFile *repo.IndexFile, kc *kudo.Client, options *Options) error {
+// getPackageCRDs tries to look for package files resolving the framework name to:
+// - a local tar.gz file
+// - a local directory
+// - a framework name in the remote repository (default)
+// in that order. Should there exist a local folder e.g. `cassandra` it will take precedence
+// over the remote repository package with the same name.
+func getPackageCRDs(name string, options *Options, repository repo.FrameworkRepository) (*repo.PackageCRDs, error) {
+	// Local files/folder have priority
+	if _, err := os.Stat(name); err == nil {
+		return repo.ReadFileSystemPackage(name)
+	}
+
+	// Construct the package name and downloan the package from the remote repo
+	indexFile, err := repository.DownloadIndexFile()
+	if err != nil {
+		return nil, errors.WithMessage(err, "could not download repository index file: %v")
+	}
 
 	var bundleVersion *repo.BundleVersion
+
 	if options.PackageVersion == "" {
 		bv, err := indexFile.GetByName(name)
 		if err != nil {
-			return errors.Wrapf(err, "getting %s in index file", name)
+			return nil, errors.Wrapf(err, "getting %s in index file", name)
 		}
 		bundleVersion = bv
 	} else {
 		bv, err := indexFile.GetByNameAndVersion(name, options.PackageVersion)
 		if err != nil {
-			return errors.Wrapf(err, "getting %s in index file", name)
+			return nil, errors.Wrapf(err, "getting %s in index file", name)
 		}
 		bundleVersion = bv
 	}
 
 	packageName := bundleVersion.Name + "-" + bundleVersion.Version
 
-	crds, err := repository.GetPackage(packageName)
+	return repository.GetPackage(packageName)
```

Comment:
why are we calling it `GetPackages` when we are actually downloading the tgz file?

We should probably update the comment over the function that states `from the given repo` (there is no repo given).

Reply:
> why are we calling it GetPackages when we are actually downloading the tgz file?

because it not only downloads the tar.gz but also unmarshals it into the `PackageCRDs` struct. I'm open to better names though